types/calidation/index.d.ts

```diff
@@ -1,4 +1,4 @@
-// Type definitions for calidation 1.16
+// Type definitions for calidation 2.0
```

Comment:
I think this needs to be bumped since it's a breaking change

Reply:
When this gets published, will it be usable with 1.16 though? Or will Typescript say that the versions don't match and are incompatible?

Another thought would be for you to fork Calidation and re-write it in Typescript with these generics improvements in mind. There are plenty of 2.0 roadmap features you could incorporate as well if interested.

Reply:
It is compatible with calidation 1.16, just that you can't update your types and expect no breaking changes. I might misunderstand when you need to bump versions of the types. Is it if the types have a breaking change, or the library itself? Both?

Reply:
I think (but don't know for sure), this will publish types v2, which is good so we don't mess up anybody using types v1.16. But I am afraid that there will be a mismatch in Typescript, when you try to install types v2 while using calidation v1.16.2

I haven't tried this scenario which is why I'm being overly cautious. If you feel confident that everything will work out as intended I'd say go ahead with the publish, watch and wait for it, check the version, install it and try to run it with the latest version of the library and be prepared to roll things back just in case.

Reply:
It should work but according to this thread https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34047 it seems breaking changes to types for a particular version of a package are to be treated as a patch. There seems to be a bit of confusion in the community about this semver vs discoverability approach. I understand both arguments. For this change I will just revert the version bump as that seems to be in accordance with how things are done in this project.